app/controllers/admin/purchases_controller.rb

```diff
@@ -52,8 +52,10 @@ def resend_receipt
       @purchase.resend_receipt
       render json: { success: true }
     else
-      render json: { success: false }
+      render json: { success: false }, status: :not_found
     end
+  rescue ActiveRecord::RecordInvalid => e
+    render json: { success: false, message: e.message }, status: :unprocessable_content
```

Comment:
In case `@purchase.save!` fails, we need to catch it so we have an error message on the UI instead of a silent fail